src/basic/unit-name.c

```diff
@@ -736,16 +738,23 @@ int unit_name_mangle_with_suffix(
                 suggest_escape = false;
         }
 
-        if (is_device_path(name)) {
-                r = unit_name_from_path(name, ".device", ret);
-                if (r >= 0)
-                        return 1;
-                if (r != -EINVAL)
+        if (path_is_absolute(name)) {
+                _cleanup_free_ char *t = NULL;
+
+                r = chase(name, NULL, CHASE_NOFOLLOW|CHASE_NONEXISTENT|CHASE_NO_AUTOFS|CHASE_WARN, &t, NULL);
```

Comment:
This is a library like function. I'd like to drop `CHASE_WARN`.

Reply:
There's a `flags` argument. Now `CHASE_WARN` is conditionalized on `flags`.